frontend/catalyst/pennylane_extensions.py

```diff
@@ -395,15 +400,27 @@ def __call__(self, *args, **kwargs):
 
         if EvaluationContext.is_tracing():
             fn = _ensure_differentiable(self.fn)
+            argnum_is_int = (
+                isinstance(self.grad_params.argnum, int) or self.grad_params.argnum is None
+            )
             grad_params = _check_grad_params(*self.grad_params)
             jaxpr, out_tree = _make_jaxpr_check_differentiable(fn, grad_params, *args)
 
-            args_data, _ = tree_flatten(args)
+            args_data, in_tree = tree_flatten(args)
+
+            total_argnums = list(range(0, len(args_data)))
+            argnums_unflatten = tree_unflatten(in_tree, total_argnums)
+            argnums_selected = [argnums_unflatten[i] for i in grad_params.argnum]
+            argnums_correct, _ = tree_flatten(argnums_selected)
+            args_argnum = tuple(args[i] for i in grad_params.argnum)
+            _, in_tree = tree_flatten(args_argnum)
+            grad_params.argnum = argnums_correct
```

Comment:
Would it make sense to move the argnum processing straight into `_check_grad_params` (would apply to jvp/vjp as well)?

Reply:
Let me give a try